barretenberg/cpp/src/barretenberg/bb/main.cpp

```diff
@@ -38,6 +39,28 @@ acir_format::acir_format get_constraint_system(std::string const& bytecode_path)
     return acir_format::circuit_buf_to_acir_format(bytecode);
 }
 
+std::vector<uint8_t> read_public_inputs(std::string const& public_inputs_path, size_t num_public_inputs)
+{
+    // If the number of public inputs is 0, then read_file will trigger a failure
+    // because the file will be empty.
+    if (num_public_inputs == 0) {
+        return {};
+    }
+    return read_file(public_inputs_path);
+}
+
+// When given a proof path, we simply append "-public_inputs" to it.
+// to derive a file path for the public inputs.
+//
+// The alternative is to have the user pass in a path for their proof
+// and a path for their public inputs. This is more verbose and
+// its likely that the user will want their public inputs to be
+// in the same directory as their proof.
+std::string public_inputs_path_from_proof_path(std::string const& proof_path)
+{
+    return proof_path + "-public_inputs";
```

Comment:
Should we mix `-` and `_` like this? If we generally use `_` as our separator lets just make this `_public_inputs`.